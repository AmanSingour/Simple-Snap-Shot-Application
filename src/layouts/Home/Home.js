import React from "react";
import { AuthContext } from "../../services/context";

import { ImageView, SearchBar } from "../../components";

import style from "./style.module.css";
import { useSelector } from "react-redux";
import { _routes } from "../../utils";
import history from "../../config/history";

//? THIS IS HOME PAGE ONLY ACCESSABLE BY AUTHENTICATED USERS
export const Home = () => {

  const photos = useSelector(state => state.photos)

  return (
    <>
      <div className={style.Container}>
        <AuthContext.Consumer>
          {(user) => user? <h1 className={style.Heading}>Welcome {user.username} to Snap-Shot!</h1>:
            history.push(_routes.LOGIN)
          }
        </AuthContext.Consumer>

        <SearchBar />
        <ImageView data={photos} />
      </div>
    </>
  );
};

export default Home;
