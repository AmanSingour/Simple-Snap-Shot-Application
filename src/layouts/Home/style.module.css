.center, .Container {
  margin: 0 auto;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  -webkit-box-pack: center;
      -ms-flex-pack: center;
          justify-content: center;
  text-align: center;
}

.vertical-center {
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, 50%);
  -webkit-transform: translate(-50%, -50%);
}

.Container {
  padding: 3rem;
}
/*# sourceMappingURL=style.module.css.map */