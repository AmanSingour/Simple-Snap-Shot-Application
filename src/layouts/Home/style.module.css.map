{
    "version": 3,
    "mappings": "ACIA,AAAA,OAAO,EDFP,UAAU,CCEH;EACH,MAAM,EAAE,MAAM;EACd,WAAW,EAAE,MAAM;EACnB,eAAe,EAAE,MAAM;EACvB,UAAU,EAAE,MAAM;CACrB;;AAED,AAAA,gBAAgB,CAAA;EACZ,QAAQ,EAAE,QAAQ;EAClB,GAAG,EAAE,GAAG;EACR,IAAI,EAAE,GAAG;EACT,SAAS,EAAE,oBAAoB;EAC/B,iBAAiB,EAAE,qBAAqB;CAC3C;;ADfD,AAAA,UAAU,CAAA;EAEN,OAAO,EAAE,IAAI;EACb,MAAM,EAAE,KAAK;EACb,UAAU,EAAE,KAAK;CACpB",
    "sources": [
        "style.module.scss",
        "../../_common.scss",
        "../../utils/theme/_sizes.scss",
        "../../utils/theme/_breakpoints.scss",
        "../../utils/theme/_colors.scss"
    ],
    "names": [],
    "file": "style.module.css"
}