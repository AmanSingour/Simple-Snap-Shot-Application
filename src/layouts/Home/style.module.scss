@import '../../common';

.Container{
    @extend .center;
    padding: 3rem;
    height: 100vh;
    max-height: 100vh;
}