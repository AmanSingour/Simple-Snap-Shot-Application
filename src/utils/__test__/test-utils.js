import React from "react";
import { Provider } from "react-redux";
import { createStore } from "redux";
import reducers from "../../services/reducers";

import { render as rtlRender } from "@testing-library/react";

// REDUX PROVIDER SETUP

function render(
  ui,
  {
    initialState,
    store = createStore(reducers, initialState),
    ...renderOptions
  } = {}
) {

  function Wrapper({ children }) {
    return <Provider store={store}>{children}</Provider>;
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

export * from "@testing-library/react";

export { render };
