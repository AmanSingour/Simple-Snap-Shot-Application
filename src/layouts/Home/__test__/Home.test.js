///import { expect } from '@jest/globals'
import { render } from "../../../utils/__test__/test-utils";
import Home from "./../Home";



describe("***Home Page running status***", () => {
  let wrapper;

  beforeEach(() => (wrapper = render(<Home />, { initialState: {} })));

  it("-Run without context", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('-run with context', ()=>{
    
  })

});
