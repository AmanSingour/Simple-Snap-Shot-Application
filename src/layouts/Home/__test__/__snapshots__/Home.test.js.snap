// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`***Home Page running status*** -Run without context 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="Container"
      >
        <div
          class="Container"
        >
          <div
            class="ui fluid icon input"
          >
            <input
              id="search"
              placeholder="Search..."
              type="text"
            />
            <i
              aria-hidden="true"
              class="search icon"
            />
          </div>
          <div
            class="Labels"
          >
            <a
              class="ui large label Label"
            >
              Sunset
            </a>
            <a
              class="ui large label Label"
            >
              Mountain
            </a>
            <a
              class="ui large label Label"
            >
              Bike
            </a>
            <a
              class="ui large label Label"
            >
              Cat
            </a>
            <a
              class="ui large label Label"
            >
              Bird
            </a>
          </div>
        </div>
        <div>
          <div />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="Container"
    >
      <div
        class="Container"
      >
        <div
          class="ui fluid icon input"
        >
          <input
            id="search"
            placeholder="Search..."
            type="text"
          />
          <i
            aria-hidden="true"
            class="search icon"
          />
        </div>
        <div
          class="Labels"
        >
          <a
            class="ui large label Label"
          >
            Sunset
          </a>
          <a
            class="ui large label Label"
          >
            Mountain
          </a>
          <a
            class="ui large label Label"
          >
            Bike
          </a>
          <a
            class="ui large label Label"
          >
            Cat
          </a>
          <a
            class="ui large label Label"
          >
            Bird
          </a>
        </div>
      </div>
      <div>
        <div />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`***Home Page running status*** renders without crushing 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="Container"
      >
        <div
          class="Container"
        >
          <div
            class="ui fluid icon input"
          >
            <input
              id="search"
              placeholder="Search..."
              type="text"
            />
            <i
              aria-hidden="true"
              class="search icon"
            />
          </div>
          <div
            class="Labels"
          >
            <a
              class="ui large label Label"
            >
              Sunset
            </a>
            <a
              class="ui large label Label"
            >
              Mountain
            </a>
            <a
              class="ui large label Label"
            >
              Bike
            </a>
            <a
              class="ui large label Label"
            >
              Cat
            </a>
            <a
              class="ui large label Label"
            >
              Bird
            </a>
          </div>
        </div>
        <div>
          <div />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="Container"
    >
      <div
        class="Container"
      >
        <div
          class="ui fluid icon input"
        >
          <input
            id="search"
            placeholder="Search..."
            type="text"
          />
          <i
            aria-hidden="true"
            class="search icon"
          />
        </div>
        <div
          class="Labels"
        >
          <a
            class="ui large label Label"
          >
            Sunset
          </a>
          <a
            class="ui large label Label"
          >
            Mountain
          </a>
          <a
            class="ui large label Label"
          >
            Bike
          </a>
          <a
            class="ui large label Label"
          >
            Cat
          </a>
          <a
            class="ui large label Label"
          >
            Bird
          </a>
        </div>
      </div>
      <div>
        <div />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
